### PR TITLE
build: pandas min version

### PR DESCRIPTION
**Pull request by nstarman**
See https://github.com/astropy/astropy/pull/15849#discussion_r1569431532

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
With v6.1rc out, we don't backport to v6.0.x anymore, so I adjusted the milestone. FYI.

Also, need to update this:

https://github.com/astropy/astropy/blob/7542bc9699cb8f022619db7ac31196b23f0b114d/tox.ini#L80

**Comment by nstarman** [comment]
It would be nice if we could tie oldest-deps in tox to our pyproject.

**Comment by pllim** [comment]
> It would be nice if we could tie oldest-deps in tox to our pyproject.

@WilliamJamieson might have wrote something up for such things at one point but I don't remember where. I vaguely remember @jdavies-st had ideas too. 🤞

**Comment by pllim** [comment]
From a quick grep, we also don't need this anymore:

https://github.com/astropy/astropy/blob/7542bc9699cb8f022619db7ac31196b23f0b114d/astropy/table/table.py#L95

Also the blurb about "Version 0.14" in `docs/install.rst` under `pandas`.

Maybe can also get rid of this but @taldcroft should confirm:

https://github.com/astropy/astropy/blob/7542bc9699cb8f022619db7ac31196b23f0b114d/astropy/table/table.py#L4147-L4154

**Comment by jdavies-st** [comment]
> > It would be nice if we could tie oldest-deps in tox to our pyproject.
> 
> @WilliamJamieson might have wrote something up for such things at one point but I don't remember where. I vaguely remember @jdavies-st had ideas too. 🤞

https://github.com/spacetelescope/minimum_dependencies is the package you want.  It was originally a script within `jwst`, but William pulled it out into its own package.  It parses the minimum versions of packages specified in `pyproject.toml` and then writes out a `requirements.txt` file with these versions pinned.  The CI job can then install these versions first, then the package without deps, and see whether the tests pass.

**Comment by pllim** [comment]
But @taldcroft , can we remove the snippet mentioned here?

https://github.com/astropy/astropy/pull/16308#issuecomment-2064715107

**Comment by taldcroft** [comment]
> But @taldcroft , can we remove the snippet mentioned here?
> 
> [#16308 (comment)](https://github.com/astropy/astropy/pull/16308#issuecomment-2064715107)

Yes, that can go since pandas >= 0.24 is always true. The one thing is that this removes the warning if the user specifies `use_nullable_int = False` in the call to `to_pandas()`. I think this is reasonable and a good thing -- the user should not to be warned about something they specifically opted into. The doc strings needs to be updated.
```
        use_nullable_int : bool, default=True
            Convert integer MaskedColumn to pandas nullable integer type.
            If ``use_nullable_int=False`` and there are masked elements then
            the column is converted to float with NaN for missing elements.
```
This is a minor API change but a good cleanup for 7.0. In a perfect world this change will break a test (for the warning) but I'm not sure off-hand.

**Comment by nstarman** [comment]
pre-commit.ci autofix

**Comment by nstarman** [comment]
Hm. Appears that pandas v1.5.3 doesn't raise the same error as later versions, instead emitting a warning.

**Comment by nstarman** [comment]
@taldcroft that fix doesn't quite work...

**Review by taldcroft — APPROVED** [review]
LGTM!

File: pyproject.toml [hunk 1]
```diff
@@ -81,7 +81,7 @@ all = [
     "beautifulsoup4",
     "html5lib",
     "bleach",
-    "pandas",
+    "pandas>=2.0",
```
Comment (pllim) [inline pyproject.toml:84]: Do we need a change log? 🤔

Comment (nstarman) [reply pyproject.toml:84]: Do we have one for optional dependency bumps following our support cadence? We should maybe standardize this decision and update our what's new template to reflect this?

Comment (pllim) [reply pyproject.toml:84]: I am not sure but suddenly going from no pin to 2.0 might surprise some users. I think `pandas` is popular enough nowadays that we should probably announce the pin? What do you think, @taldcroft ?

Comment (nstarman) [reply pyproject.toml:84]: Ping @taldcroft.

Comment (taldcroft) [reply pyproject.toml:84]: It might be a good idea here to add a changelog entry esp. since this is a major version update for pandas.

**Comment by taldcroft** [comment]
Looks like RTD is failing, otherwise this LGTM.

**Comment by nstarman** [comment]
Rebased.

**Comment by pllim** [comment]
Please rebase one more time. There was one more doc fix unmerged. Thanks for your patience!

**Pull request merged**